plasma_framework/contracts/mocks/utils/RLPMock.sol

```diff
@@ -17,29 +19,96 @@ contract RLPMock {
         return bytes20(_data.toRlpItem().toAddress());
     }
 
-    function decodeInt(bytes memory _data) public pure returns (int) {
-        return int(_data.toRlpItem().toUint());
+    function decodeBytes(bytes memory _data) public pure returns (bytes memory) {
+        return toBytes(_data.toRlpItem());
     }
 
     function decodeUint(bytes memory _data) public pure returns (uint) {
         return _data.toRlpItem().toUint();
     }
 
+    function decodePayloadOffset(bytes memory _data) public pure returns (uint) {
+        _data.toRlpItem().decodePayloadOffset();
+    }
+
+    function decodeInt(bytes memory _data) public pure returns (int) {
+        return int(_data.toRlpItem().toUint());
+    }
+
+    function decodeString(bytes memory _data) public pure returns (string memory) {
+        return string(toBytes(_data.toRlpItem()));
+    }
+
     function decodeList(bytes memory _data) public pure returns (bytes[] memory) {
+
         RLPReader.RLPItem[] memory items = _data.toRlpItem().toList();
 
         bytes[] memory result =  new bytes[](items.length);
         for (uint i = 0; i < items.length; i++) {
-            result[i] = items[i].toRlpBytes();
+            result[i] = toRlpBytes(items[i]);
         }
         return result;
     }
 
-    function decodeString(bytes memory _data) public pure returns (string memory) {
-        return string(_data.toRlpItem().toBytes());
+    function toBytes(RLPReader.RLPItem memory item) internal pure returns (bytes memory) {
```

Comment:
I am reading this as moving codes we don't need yet to mock contract. Is this correct? If that is the case, can you add comments on why we are having such complex functions in our mock.

Reply:
I wanted to make it clear on what parts of the code are used in production and which ones are not. Therefore I moved everything that is not strictly necessary to the mock contract. 

We have them so after the RLP has been decoded we can compare them back to strings and bytes. Where should I add a comment in the mock class?

Reply:
Probably just on top of the contract and explain a bit. I am asking this as most our mock are used for:
1. Purely proxy the call to implementation
1. Add some simple helper function to inject states

So this is a bit out. I feel we are putting here to conform the solidity spec tests (?) and just for conveniency if we ever need reference code, right?

Reply:
Yes solidity spec tests is one reason, the other is that the code is unsafe in some of this functions. This can be distracting for the audit and in general for running automated security analysis tools that would report issues.